llvm/include/llvm/Support/GenericDomTreeConstruction.h

```diff
@@ -180,15 +180,17 @@ struct SemiNCAInfo {
                   unsigned AttachToNum,
                   const NodeOrderMap *SuccOrder = nullptr) {
     assert(V);
-    SmallVector<NodePtr, 64> WorkList = {V};
+    SmallVector<std::pair<NodePtr, unsigned>, 64> WorkList = {{V, AttachToNum}};
```

Comment:
This looks like a pretty large stack allocation (16B * 64 => 1kB?). Is this what we want?

Reply:
It's on the bigger side, but as runDFS is essentially a leaf function, I don't think it's a problem.